### DISCUSSION TITLE
[Q&A] How to identify the peer in task_data_filters on the Controller

### DISCUSSION THREAD

**Discussion by evgvain** [NVIDIA/NVFlare#2961] [Q&A] [answered]
### Python version (`python3 -V`)

3.10

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.5

### NVFlare branch (if running examples, please use the branch that corresponds to the NVFlare version, `git branch`)

2.5

### Operating system

Ubuntu 22.04

### Have you successfully run any of the following examples?

- [X] hello-numpy-sag with simulator
- [X] hello-pt with simulator
- [X] hello-numpy-sag with POC
- [X] hello-pt with POC

### Please describe your question

The logic of my server's task_data_filter implementation depends on what specif client the task data is sent to.
I am looking for a correct way to figure out the context of what peer the actual filter is invoked.
The only way I discovered is to get it in DXOFilter from the FLContext: 
`peer = fl_ctx.get_peer_context().props[ReservedKey.CLIENT_NAME]['value']`
Is this this a right approach?

**Comment by YuanTingHsieh** [comment, marked as answer]
Hi @evgvain thanks for the question!

Have you tried: `fl_ctx.get_peer_context().get_identity_name()`?

**Reply by evgvain** [reply]
Thanks, it works!
